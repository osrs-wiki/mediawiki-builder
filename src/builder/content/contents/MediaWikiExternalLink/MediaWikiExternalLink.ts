import MediaWikiContent from "../../MediaWikiContent";
import { MediaWikiContents } from "../../MediaWikiContent.types";

export class MediaWikiExternalLink extends MediaWikiContent {
  link: string;

  constructor(label: MediaWikiContents, link: string) {
    super(label);
    this.link = link;
  }

  build() {
    return `[${this.link} ${this.buildChildren()}]`;
  }
}
