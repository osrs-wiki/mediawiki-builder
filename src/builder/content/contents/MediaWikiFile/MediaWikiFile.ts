import { MediaWikiFileOptions } from "./MediaWikiFile.types";
import MediaWikiContent from "../../MediaWikiContent";
import { buildContents } from "../../MediaWikiContent.utils";

export class MediaWikiFile extends MediaWikiContent {
  fileName: string;
  options?: MediaWikiFileOptions;

  constructor(fileName: string, options?: MediaWikiFileOptions) {
    super();
    this.fileName = fileName;
    this.options = options;
  }

  build() {
    let options = "";
    const {
      caption,
      format,
      resizing,
      horizontalAlignment,
      verticalAlignment,
      link,
    } = this.options ?? {};
    if (format) {
      options += `|${format}`;
    }
    if (resizing?.width && resizing?.height) {
      options += `|${resizing.width}x${resizing.height}px`;
    } else if (resizing?.width) {
      options += `|${resizing.width}px`;
    } else if (resizing?.height) {
      options += `|${resizing.height}px`;
    }
    if (horizontalAlignment) {
      options += `|${horizontalAlignment}`;
    }
    if (verticalAlignment) {
      options += `|${verticalAlignment}`;
    }
    if (caption) {
      options += `|${buildContents(caption)}`;
    }
    if (link) {
      options += `|link=${link}`;
    }
    return `[[File:${this.fileName}${options}]]`;
  }
}
