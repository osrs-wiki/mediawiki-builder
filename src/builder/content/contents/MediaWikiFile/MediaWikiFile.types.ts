import { MediaWikiContents } from "../../MediaWikiContent.types";

export type FileFormat = "frameless" | "frame" | "thumb";

export type FileHorizontalAlignment = "left" | "right" | "center" | "none";

export type FileVerticalAlignment =
  | "baseline"
  | "sub"
  | "super"
  | "top"
  | "text-top"
  | "bottom"
  | "text-bottom";

export type FileResizing = {
  width?: number;
  height?: number;
};

export type MediaWikiFileOptions = {
  caption?: MediaWikiContents;
  format?: FileFormat;
  resizing?: FileResizing;
  horizontalAlignment?: FileHorizontalAlignment;
  verticalAlignment?: FileVerticalAlignment;
  link?: string;
};
