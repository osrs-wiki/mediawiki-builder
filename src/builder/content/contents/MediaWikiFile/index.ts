export * from "./MediaWikiFile";
export * from "./MediaWikiFile.types";
