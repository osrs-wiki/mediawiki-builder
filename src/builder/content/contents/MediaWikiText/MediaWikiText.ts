import { MediaWikiTextStyling } from "./MediaWikiText.types";
import MediaWikiContent from "../../MediaWikiContent";
import { MediaWikiContents } from "../../MediaWikiContent.types";

export class MediaWikiText extends MediaWikiContent {
  styling?: MediaWikiTextStyling;

  constructor(value: MediaWikiContents, styling?: MediaWikiTextStyling) {
    super(value);
    this.styling = styling;
  }

  build() {
    let builtValue = this.buildChildren();
    if (builtValue === "") {
      return "";
    }
    if (this.styling?.bold) {
      builtValue = `'''${builtValue}'''`;
    }
    if (this.styling?.italics) {
      builtValue = `''${builtValue}''`;
    }
    if (this.styling?.underline) {
      builtValue = `<u>${builtValue}</u>`;
    }
    return `${builtValue}`;
  }
}
