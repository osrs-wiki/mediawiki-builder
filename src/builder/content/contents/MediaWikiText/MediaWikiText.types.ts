export type MediaWikiTextStyling = {
  italics?: boolean;
  bold?: boolean;
  underline?: boolean;
};
