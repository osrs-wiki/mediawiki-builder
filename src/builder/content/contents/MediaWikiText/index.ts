export * from "./MediaWikiText";
export * from "./MediaWikiText.types";
