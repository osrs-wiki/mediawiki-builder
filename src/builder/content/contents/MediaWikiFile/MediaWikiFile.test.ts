import { MediaWikiFile } from "./MediaWikiFile";
import { MediaWikiExternalLink } from "../MediaWikiExternalLink";
import { MediaWikiText } from "../MediaWikiText";

describe("MediaWikiFile", () => {
  test("caption with MediaWikiText", () => {
    expect(
      new MediaWikiFile("test.png", {
        caption: new MediaWikiText("caption", { italics: true }),
      }).build()
    ).toMatchSnapshot();
  });

  test("caption with MediaWikiContent array", () => {
    expect(
      new MediaWikiFile("test.png", {
        caption: [
          new MediaWikiText("caption ", { italics: true }),
          new MediaWikiExternalLink("link", "https://test.com"),
          new MediaWikiText(" test", { italics: true }),
        ],
      }).build()
    ).toMatchSnapshot();
  });
});
