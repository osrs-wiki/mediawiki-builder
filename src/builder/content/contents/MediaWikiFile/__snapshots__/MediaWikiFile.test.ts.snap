// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaWikiFile caption with MediaWikiContent array 1`] = `"[[File:test.png|''caption ''[https://test.com link]'' test'']]"`;

exports[`MediaWikiFile caption with MediaWikiText 1`] = `"[[File:test.png|''caption'']]"`;
